photon-ml/src/main/scala/com/linkedin/photon/ml/supervised/classification/SmoothedHingeLossLinearSVMModel.scala

```diff
@@ -69,3 +70,14 @@ class SmoothedHingeLossLinearSVMModel(override val coefficients: Coefficients)
     }
   }
 }
+
+object SmoothedHingeLossLinearSVMModel {
+  /**
+   * Create a new smoothed hinge loss SVM model with the provided coefficients (means) and variances
+   *
+   * @param coefficients The feature coefficient means and variances for the model
+   * @return A smoothed hinge loss SVM model
+   */
+  def createModel(coefficients: Coefficients): SmoothedHingeLossLinearSVMModel =
+    new SmoothedHingeLossLinearSVMModel(coefficients)
```

Comment:
For the regression model classes (LinearRegressionModel, LogisticRegressionModel, etc.) it's necessary to keep the GeneralizedLinearOptimizationProblem and descendants simple and avoid the type/covariance issues we had prior to the refactor.

For the OptimizerFactory, it's a standard implementation of the Factory Pattern: we don't know what kind of Optimizer to generate until runtime. 

For the optimization problem classes it's just a convenience method, it's not necessary. In these cases the factory methods serve as constructors.

After I noticed that I was using the factory pattern, I decided I may as well restrict object construction for those classes to the factory method. I don't have any hard reasoning, it just seemed natural to fully commit to the factory pattern, also to limit access as strictly as possible and then loosen it if necessary.